### ISSUE TITLE
.gif image displaying is getting messed up for me.

### ISSUE THREAD

**Issue by o355** [jarvisteach/appJar#160]
Hey, loving AppJar. I'm running into a big obstacle with .gif images, and how they're appearing.

Let's take this .gif file, a radar loop of NYC from 100 km: http://imgur.com/a/HyD4v

On my end, with my GUI, the side of the .gif is "blanking out" on the right side of the gif. Worse, when I fetch a more zoomed in radar loop with nothing going on (except the time changing), I can only see the time changes, nothing else. Here's an example: http://imgur.com/a/oqMaa

However, when lots of stuff is happening on screen, the .gif plays seamlessly.

I've tried basic troubleshooting, doing image clears, all that fun stuff. Nothing seems to be working. I was really looking forward to using appJar to display radar images for a program I'm making, but this really hinders the radar viewer component of my problem.

No errors are getting outputted to the console. I'm running Windows 10 with Python 3.6 as my development environment. I have an Ubuntu 16.04.2 LTS box with Python 3.5, so I'll test my script on there soon.

Anyways, thanks for the great library. I've enjoyed using appJar to make GUIs!

**Comment by jarvisteach** [comment]
Just had a quick look - same issues on Mac OSX. I've coded in support for animated gifs myself, and it all looks to be working fine. The image gets identified as an animated gif, the dimensions are read correctly, and each of the image files seems to get loaded OK. But, for whatever reason, the individual frames seem to be getting cropped down that right-hand side.

Loading each frame is done with the code: `PhotoImage(file=img.path, format="gif - {0}".format(img.anim_pos))`

When tested outside of appJar, with some simple tkinter code, I'm seeing the same result:

```python
from tkinter import *
root = Tk()

def update(x):
    img = PhotoImage(file='radar.gif',format='gif -index '+str(x))
    x += 1
    if x > 9: x = 0
    label.configure(image=img)
    label.img=img
    root.after(100, update, x)

label = Label(root)
label.pack()
root.after(0, update, 0)
root.mainloop()
```

So, it's not an issue with appJar, but either the image files, or the tkinter PhotoImage library.

I've tried testing with another image: https://s-media-cache-ak0.pinimg.com/originals/60/ea/16/60ea1674d2a2221937f5171f21aec302.gif

And I see some interesting results - again the image is being cropped, but on all sides. It looks, perhaps, that animated gifs don't necessary store all pixels in each frame, but rather rely on the app to fill in the blanks when pixels (eg. a background) are repeated across multiple frames. This would make sense, as it would reduce the file size.

I'll need to do some investigation into the PhotoImage class of tkinter, to see if there is some flag I'm not setting to resolve this...

**Comment by jarvisteach** [comment]
Yup, according to the GIF wikipedia page: https://en.wikipedia.org/wiki/GIF#Animated_GIF

Animated GIFs can specify just the portion that has changed for each frame - explaining why you only see the time changes.

So, need to find a way of filling in the gaps from the previous frames...

I imagine, that if appJar used PIL, this might not be an issue...

**Comment by jarvisteach** [comment]
https://gist.github.com/almost/d2832d0998ad9dfec2cacef934e7d247
